src/api/MetadataQuery.js

```diff
@@ -0,0 +1,115 @@
+     * @param {List<ConditionType>} conditions - the list of query conditions
+     * @param {ColumnType} orderByColumn - the column to order by
+     * @param {SortOrderType} orderDirection - the typed sort order ('ASC', 'DESC')
+     * @param {ConnectorType} connector connector for query ('AND' / 'OR')
+     * @return {void}
+     */
+
+    createQueryPostBody = (
+        scope: string,
+        templateKey: string,
+        columns: List<ColumnType>,
+        conditions: List<ConditionType>,
+        orderByColumn: ColumnType,
+        orderDirection: SortOrderType,
+        connector: ConnectorType,
+    ) => {
+        const queryParams = {}; // Hashmap of arguments used in conditions
+        const conditionsList = []; // Array of conditions joined (by connector) to form the final query predicate
+
+        conditions.forEach(({ columnId, operator: operatorSymbol, values }, conditionIndex) => {
+            const targetColumn = columns.filter(column => column.id === columnId).get(0);
+            if (!targetColumn) {
+                return;
+            }
+            const { property } = targetColumn;
+            conditionsList.push(`${scope}.${templateKey}.${property} ${operatorSymbol} :arg_${conditionIndex}`);
+
+            // If the value can be interpreted as numeric, the field is likely numeric
+            // TODO: Include field type in columns to avoid this potentially incorrect assumption
+            const parsedValue = isNaN(values[0]) ? values[0] : parseFloat(values[0]);
+
+            queryParams[`arg_${conditionIndex}`] = parsedValue; // TODO: Support multi-value conditions
+        });
+
+        let query = conditionsList.join(` ${connector} `);
+        let orderBy = [];
+        if (query) {
+            const { property } = orderByColumn;
+            orderBy = [{ field_key: `${scope}.${templateKey}.${property}`, direction: orderDirection }];
+        } else {
+            query = `INSTANCE_EXISTS(${scope}.${templateKey})`;
+        }
+
+        return {
+            query,
+            query_params: queryParams,
+            order_by: orderBy,
+        };
+    };
+
+    /**
+     * Gets query results
+     *
+     * @param {string} ancestorFolderId - ancestor folder id
+     * @param {Object} queryPostBody - query object built using createQueryPostBody
+     * @param {number} limit - maximum number of items to retrieve
+     * @param {string} marker - marker string for marker-based pagination
+     * @param {Function} successCallback - Function to call with results
+     * @param {Function} errorCallback - Function to call with errors
+     * @return {void}
+     */
+    query(
+        ancestorFolderId: string,
+        queryPostBody: Object,
+        limit: number,
+        marker: string,
+        successCallback: Function,
+        errorCallback: ElementsErrorCallback,
+    ): void {
+        this.errorCode = ERROR_CODE_METADATA_QUERY;
+
+        this.post({
+            id: ancestorFolderId,
```

Comment:
What's the purpose of id? 

If needed to distinguish between queries, `ancestoryFolderId` is not sufficient.